corehq/apps/case_importer/tracking/models.py

```diff
@@ -0,0 +1,41 @@
+from django.db import models
+from dimagi.ext import jsonobject
+from dimagi.utils.decorators.memoized import memoized
+from soil.util import get_task
+
+
+class CaseUploadRecord(models.Model):
+    domain = models.CharField(max_length=256)
+    created = models.DateTimeField(auto_now_add=True)
+
+    upload_id = models.UUIDField(unique=True)
+    task_id = models.UUIDField(unique=True)
+    couch_user_id = models.CharField(max_length=256)
+    case_type = models.CharField(max_length=256)
+
+    @property
+    @memoized
+    def task(self):
+        return get_task(self.task_id)
+
+
+class CaseUploadJSON(jsonobject.JsonObject):
```

Comment:
Interested to hear why you chose this approach to producing JSON instead of having a `to_json` method on the `CaseUploadRecord` that either just creates a dict or uses a serializer to create the dict.